tcpdump.c

```diff
@@ -844,9 +844,24 @@ MakeFilename(char *buffer, char *orig_name, int cnt, int max_chars)
           }
 
           /* There's no good way to detect an error in strftime since a return
-           * value of 0 isn't necessarily failure.
+           * value of 0 isn't necessarily failure; if orig_name is an empty
+           * string, the formatted string will be empty.
+           *
+           * However, the C90 standard says that, if there *is* a
+           * buffer overflow, the content of the buffer is undefined,
+           * so we must check for a buffer overflow.
+           *
+           * So we check for an empty orig_name, and only call
+           * strftime() if it's non-empty, in which case the return
+           * value will only be 0 if the formatted date doesn't fit
+           * in the buffer.
            */
-          strftime(filename, PATH_MAX, orig_name, local_tm);
+          if (strlen(orig_name) != 0) {
+	    if (0 == strftime(filename, PATH_MAX, orig_name, local_tm))
```

Comment:
Style: we have `xxx != 0`, we should have `yyy == 0`, not `0 == yyy`

Reply:
Done.